File: leetCode/01_shuffle_string.js
const shuffleString = (s, indices) => {
    let resultArray = new Array(s.length)
    indices.forEach((value, i) => {
        resultArray[value] = s[i]
    });
    return resultArray.join("")
}
console.log(shuffleString("ohal", [1,0,3,2]))

File: leetCode/01_shuffle_string.py
from typing import List

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        result_array: List[int] = [""] * len(indices)
        for i, value in enumerate(indices):
            result_array[value] = s[i]
            
        output: str = ""
        for element in result_array:
            output += element
        return output

my_s, my_list = "codeleet", [4,5,6,7,0,2,1,3]

print(Solution.restoreString(Solution, my_s, my_list))